core-foundation/src/dictionary.rs

```diff
@@ -73,8 +73,9 @@ impl CFDictionary {
     /// Similar to `contains_key` but acts on a higher level, automatically converting from any
     /// `TCFType` to the raw pointer of its concrete TypeRef.
     #[inline]
-    pub fn contains_key2<X, K: TCFType<*const X>>(&self, key: &K) -> bool {
-        self.contains_key(key.as_concrete_TypeRef() as *const c_void)
+    pub fn contains_key2<K: TCFType>(&self, key: &K) -> bool {
+        let type_ref_ptr = &key.as_concrete_TypeRef() as *const _ as *const *const c_void;
+        self.contains_key(unsafe { *type_ref_ptr })
```

Comment:
This implementation had to pay with a bit complexity for the reduced API complexity.

Basically, now I don't know that `K::TypeRef` is a `*const _` any longer, so I can't just cast the pointer directly. Instead I have to take the pointer *to the pointer* and then cast that and lastly dereference.

I added a unit test to make sure I got it right.